src/app/dashboard/game.service.ts

```diff
@@ -0,0 +1,160 @@
+import { Injectable } from '@angular/core';
+import Konva from 'konva';
+import { BehaviorSubject } from 'rxjs';
+import { Hero } from '../core/interface/hero';
+
+const COLOR_HEALTH = {
+  SAFE: '#198754',
+  DANGER: '#dc3545',
+  BACKGROUND: '#ccc'
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GameService {
+  heroesPlaying = new BehaviorSubject<Hero[]>([]);
+  updateGame = new BehaviorSubject<boolean>(false);
+  drawAgain = false;
+
+  constructor() { }
+
+  updateHeroesPlaying(hero: Hero[], drawAgain?: boolean) {
+    this.drawAgain = !!drawAgain;
+    this.heroesPlaying.next(hero);
+  }
+
+  /**
+   * Draw a ring
+   * @param x: x position
+   * @param y: y position
+   * @param radius: radius of outer ring
+   * @param color: color ring
+   * @returns Konva Ring object
+   */
+  drawRing(x: number, y: number, radius: number, color: string): Konva.Ring {
+    return new Konva.Ring({
+      x,
+      y,
+      innerRadius: radius - 6,
```

Comment:
For your tests don't test that this returns a Ring but that radius is indeed -6.  You have a reason for it needing to be -6 and not -7.  So that is what the test should be protecting.

Or better yet your test is actually that outerRadius > innerRadius b/c your goal is just to show a ring, if ever it was 0 then or smaller there would be no ring.  By testing for > then you allow others to make it thicker and the intention of the test still works.

Reply:
Thanks for your comment. I do not have much experience in testing. I will improve myself in it.